### ISSUE TITLE
How to calculate batch of messages when sending to IoT Hub

### ISSUE THREAD

**Issue by iconics-pavels** [Azure/azure-iot-sdk-csharp#968]
- **OS, version, SKU and CPU architecture used:** <VERSION> Windows 10 Desktop x64
- **Application's .NET Target Framework :** <NETVERSION> net472
- **Device:** <DEVICETYPE> Desktop PC
- **SDK version used:** <VERSION> Microsoft.Azure.Devices.Client v1.20.1

## Description of the issue:
I am using DeviceClient with AMQP transport type and I want to send batch of the messages instead of sending one by one. Is it possible to calculate size of the batch of the messages that I am trying to send into the IoT Hub? I know there is a limitation set to 256-KB. But the limit includes also the header (Properties). I can easily calculate message's body but how I should calculate the header?

Also in the past I found a note where was reserve 1-KB for every 64-KB. Maximum size of cloud-to-device message is 64-KB. Is it still valid? Or should I do it in a different way?

In other words, I am asking whether there is something like Microsoft.Azure.EventHubs.EventDataBatch class but for the IoT Hub.

Thank you

**Comment by prmathur-microsoft** [comment]
I think what you are looking for is a way to tell the message size which is inline with this request so I'm linking it. https://github.com/Azure/azure-iot-sdk-csharp/issues/669

**Comment by RubiaL** [comment]
Thanks for the feedback. This is currently in our backlog and will consider it as part of our planning cycle

**Comment by az-iot-builder-01** [comment]
@iconics-pavels, @prmathur-microsoft, @RubiaL, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
